### ISSUE TITLE
Sign the Mac image

### ISSUE THREAD

**Issue by elharo** [spyder-ide/spyder#14375]
It need not be from an unidentified developer.

**Comment by ccordoba12** [comment]
Thanks @elharo for reporting. We'll try to fix that in our next version (4.2.1)m to be released in two weeks.

**Comment by ccordoba12** [comment]
@mrclary, could you help us with this? We can buy an Apple developer id using our Open Collectives funds.

Then it'd be a matter of following the instructions present here:

https://stackoverflow.com/questions/23824815/how-to-add-codesigning-to-dmg-file-in-mac

**Comment by mrclary** [comment]
@ccordoba12, yes, I can look into that. If you get the developer id, I'll take care of the rest.

**Comment by mrclary** [comment]
@ccordoba12, in reviewing again the link you provided, I don't think that will work for us. That will codesign the dmg file, but I don't think we have an issue mounting the disk image, rather it is opening the application once copied to the user's local computer.

However, I think that [this](https://localazy.com/blog/how-to-automatically-sign-macos-apps-using-github-actions) may work. I can handle updating the pipeline workflow, but I'll need you to do a few things first.

1. Obtain the Apple Developer subscription
2. Create the certificate and GitHub secrets (as shown in the above link)

**Comment by mrclary** [comment]
Closing in favor of #16414
